### PR TITLE
Implement proxy server support for external calls

### PR DESCRIPTION
**Pull request by Cruikshanks**
https://eaflood.atlassian.net/browse/WE-416

Having recently deployed our app to our development environment, we've been reminded that when the app needs to make a connection to the internet (for example when communicating with Dynamics) it will have to make that connection through a proxy server.

So this change updates our services that make http requests to external services to support using a proxy. The change also handles when working locally that we don't have a proxy server running.

### CODE REVIEW

File: src/services/activeDirectoryAuth.service.js [hunk 1]
```diff
@@ -16,15 +18,22 @@ module.exports = class ActiveDirectoryAuthService {
...
+    const options = url.parse(`https://${config.azureAuthHost}${config.azureAuthPath}`)
+    options.method = 'POST'
+    options.headers = {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Content-Length': Buffer.byteLength(this.queryParams)
     }
+    if (config.http_proxy) {
+      options.agent = new HttpsProxyAgent(config.http_proxy)
+    }
+    console.log(options)
```
Comment (duncannixon) [inline src/services/activeDirectoryAuth.service.js]: Is this still needed?

File: src/services/activeDirectoryAuth.service.js [hunk 2]
```diff
@@ -49,6 +58,8 @@ module.exports = class ActiveDirectoryAuthService {
 
           const token = tokenResponse.access_token
           if (token) {
+            console.log('Got a token')
```
Comment (duncannixon) [inline src/services/activeDirectoryAuth.service.js]: Ditto for these - still needed?

File: src/services/activeDirectoryAuth.service.js [hunk 3]
```diff
@@ -49,6 +58,8 @@ module.exports = class ActiveDirectoryAuthService {
...
+            console.log(token)
```
Comment (duncannixon) [inline src/services/activeDirectoryAuth.service.js]: ditto

File: src/services/dynamicsDal.service.js [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
+      'OData-MaxVersion': '4.0',
+      'OData-Version': '4.0',
+      'Accept': 'application/json',
+      'Content-Type': 'application/json; charset=utf-8',
+      'Prefer': 'odata.maxpagesize=500, odata.include-annotations=OData.Community.Display.V1.FormattedValue'
     }
+    if (config.http_proxy) {
+      options.agent = new HttpsProxyAgent(config.http_proxy)
+    }
+    console.log(options)
```
Comment (duncannixon) [inline src/services/dynamicsDal.service.js]: Ditto

**Comment by Cruikshanks** [comment]
Thanks for the feedback and awesome to get a head start on the review, but not yet completed this. Still trying to figure out why its not working in our dev environment (we now seem to be getting an auth token, but error when talking to Dynamics) hence lots of `console.log`s!

**Pull request merged**